### PR TITLE
[CARBONDATA-239] Handling the exception cases in compaction.

### PR DESCRIPTION
**Pull request by ravikiran23**
Why raise this pr?
If a compaction is triggered by the user on table1 , and other requests will go to queue. and if the compaction is failed for table1 then the requests in queue is not getting completed.

How to resolve?
If the compaction is failed then the requests in queue should continue the compaction.

if compaction is failed for table, requested by user then need to show error message in the beeline.

### CODE REVIEW

**Comment by ravikiran23** [comment]
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/200/

File: integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala [hunk 1]
```diff
@@ -590,16 +605,25 @@ object CarbonDataRDDFactory extends Logging {
...
+                    logger.error("Exception in compaction thread for table " + tableForCompaction
+                      .carbonTable.getDatabaseName + "." +
+                                 tableForCompaction.carbonTableIdentifier
+                                   .getTableName)
+                  // not handling the exception. only logging as this is not the table triggered
+                  // by user.
+                }
                 finally {
-                  // delete the compaction required file
+                  // delete the compaction required file in case of failure or success also.
```
Comment (gvramana) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: If compact file deletion fails, add failed table to skipped list.

Comment (ravikiran23) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/CarbonDataRDDFactory.scala]: fixed

**Comment by ravikiran23** [comment]
http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/248/

**Comment by gvramana** [comment]
LGTM

**Pull request closed without merging**